client/src/App.js

```diff
@@ -0,0 +1,125 @@
+import './App.css';
+import { useState } from 'react';
+import Score from './components/Score';
+import Question from './components/Question';
+import Flag from './components/Flag';
+
+function App() {
+
+  const [submitted, setSubmitted] = useState(false);
+  const [questionChosen, setQuestionChosen] = useState(null);
+  const [questionIndex, setQuestionIndex] = useState(0);
+  const [displayQuestion, setDisplayQuestion] = useState(false);
+  const [resultMessage, setResultMessage] = useState(null);
+  const [score, setScore] = useState(0);
+  const [disable, setDisable] = useState(false);
+  const [flagIndex, setFlagIndex] = useState(1);
+
+  const message = [
+    `Sorry, your answer is incorrect.`,
+    `Bravo! Your answer is correct. You are one step closer to completing the puzzle!`, 
+    `Bravo! Your answer is correct. Great job completing parts of the puzzle!`,
+    `Bravo! Your answer is correct. Congratulations! You have successfully completed the puzzle!`
+  ];
+
+  const flagPieces = document.getElementsByClassName('flag');
+
+  const generateQuestion = () => {
```

Comment:
Technically, you can use the `useCallback` hook to be a little more efficient. See: https://reactjs.org/docs/hooks-reference.html#usecallback To be honest, this is something I was always told to do, but I've been meaning to do some reading to see how much of a performance improvement this actually is.